sgt-back-end/server.js

```diff
@@ -0,0 +1,202 @@
+    res.status(400).json({
+      error: "Must include 'course', 'name', and 'score'. Please try again."
+    });
+    return false;
+  }
+  if (inputKeys.length > 3) {
+    res.status(400).json({
+      error: "Only 'course', 'name', and 'score' would be processed. Please try again."
+    });
+    return false;
+  }
+  for (const key of inputKeys) {
+    if (reqKeys.indexOf(key) === -1) {
+      res.status(400).json({
+        error: "Only 'course', 'name', and 'score' would be processed. Please try again."
+      });
+      return false;
+    }
+    if (!userInputs[key]) {
+      res.status(400).json({
+        error: `'${key}' was left blank. Please resubmit and include a value for '${key}'.`
+      });
+      return false;
+    }
+  }
+  if (isNaN(score) || Number(score) < 0) {
+    res.status(400).json({ error: `${score} is not a valid 'score'. Scores must be positive numbers.` });
+    return false;
+  } else {
+    return userInputs;
+  }
+}
+
+async function loadGrades(res) {
+  const sql = `
+    SELECT *
+    FROM "grades";
+  `;
+  const result = await db.query(sql);
+  if (result) {
+    return result.rows;
+  } else {
+    return false;
+  }
+}
+
+async function loadGradeId(res, id) {
+  const sql = `
+    SELECT *
+    FROM "grades"
+    WHERE "gradeId" = $1;
+  `;
+  const params = [id];
+  const result = await db.query(sql, params);
+  const grade = result.rows[0];
+  if (grade) {
+    return grade;
+  } else {
+    res.status(404).json({ error: `Cannot find grade with gradeId: ${id}` });
+    return false;
+  }
+}
+
+app.get('/api/grades', async (req, res) => {
+  try {
+    const result = await loadGrades();
+    if (result) {
+      res.json(result);
+    } else {
+      res.status(404).json({ error: 'Grades table is currently empty.' });
+    }
+  } catch (err) {
+    handleError(res, err);
+  }
+});
+
+app.get('/api/grades/:gradeId', async (req, res) => {
+  try {
+    const gradeId = Number(req.params.gradeId);
+    if (!evalParamId(res, gradeId)) {
+      return;
+    }
+    const grade = await loadGradeId(res, gradeId);
+    if (grade) {
+      res.json(grade);
+    }
+  } catch (err) {
+    handleError(res, err);
+  }
+});
+
+app.post('/api/grades', async (req, res) => {
+  try {
+    const userInputs = evalReqBody(req, res);
+    if (!userInputs) {
+      return;
+    }
+    const sql = `
+      INSERT INTO "grades" ("course", "name", "score")
+      VALUES ($1, $2, $3)
+      RETURNING *;
+    `;
+    const { course, name, score } = userInputs;
+    const params = [course, name, score];
+    const result = await db.query(sql, params);
+    const grade = result.rows[0];
+    if (grade) {
+      res.json(grade);
+    }
+  } catch (err) {
+    handleError(res, err);
+  }
+});
+
+app.delete('/api/grades/:gradeId', async (req, res) => {
+  try {
+    const gradeId = Number(req.params.gradeId);
+    if (!evalParamId(res, gradeId)) {
+      return;
+    }
+    const grade = await loadGradeId(res, gradeId);
```

Comment:
As we discussed, you can remove this and instead check the result from the DELETE call.